src/main/java/de/codecentric/centerdevice/util/StageUtils.java

```diff
@@ -46,12 +56,49 @@ public static void focusNextStage() {
 		}
 	}
 
+	@SuppressWarnings("unchecked")
 	public static ObservableList<Stage> getStages() {
-		return StageHelper.getStages();
+		if (stages == null) {
+			// Java 9
+			try {
+				windows = (ObservableList<Window>)Window.class.getMethod("getWindows").invoke(null);
+				stages = FXCollections.observableArrayList();
+				windows.addListener(new ListChangeListener<Window>() {
+					@Override
+					public void onChanged(javafx.collections.ListChangeListener.Change<? extends Window> c) {
+						updateStages();
```

Comment:
You're right, my bad. For some reason, i was thinking stages were _added_ to `stages` on `updateStages` but you're right of course, it replaces the content.